biz.aQute.bndlib/src/aQute/bnd/plugin/spi/SPIAnnotations.java

```diff
@@ -0,0 +1,229 @@
+package aQute.bnd.plugin.spi;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.osgi.namespace.extender.ExtenderNamespace;
+
+import aQute.bnd.annotation.spi.ServiceConsumer;
+import aQute.bnd.osgi.Analyzer;
+import aQute.bnd.osgi.Annotation;
+import aQute.bnd.osgi.Clazz;
+import aQute.bnd.osgi.Constants;
+import aQute.bnd.osgi.Descriptors.PackageRef;
+import aQute.bnd.osgi.Descriptors.TypeRef;
+import aQute.bnd.osgi.EmbeddedResource;
+import aQute.bnd.osgi.Processor;
+import aQute.bnd.service.AnalyzerPlugin;
+import aQute.bnd.version.Version;
+import aQute.bnd.version.VersionRange;
+import aQute.lib.exceptions.Exceptions;
+import aQute.lib.strings.Strings;
+
+public class SPIAnnotations implements AnalyzerPlugin {
+
+	private static final Version	SLM_VERSION			= new Version(1);
+
+	private static final String		SERVICE_CONSUMER	= "aQute/bnd/annotation/spi/ServiceConsumer";
+	private static final String		SERVICE_CONSUMERS	= "aQute/bnd/annotation/spi/ServiceConsumers";
+	private static final String		SERVICE_PROVIDER	= "aQute/bnd/annotation/spi/ServiceProvider";
+	private static final String		SERVICE_PROVIDERS	= "aQute/bnd/annotation/spi/ServiceProviders";
+
+	@Override
+	public boolean analyzeJar(Analyzer analyzer) throws Exception {
```

Comment:
Do we need some opt-in/out mechanism for the classes processed like we have to DS and CDI? It seems like we should to allow the developer to be in control.

Reply:
yup we do! Adding.

Reply:
oops I forgot this part...